Inconsistent behavior with @types when using tsc with a tsconfig

When using TypeScript 2, I get an inconsistent behavior for importing type definitions from `@types` when using `tsc` with an explicit `tsconfig.json` path versus not specifying one.

I have the following setup:

> **`package.json`**
```json
{
  "name": "typings-test",
  "devDependencies": {
    "@types/es6-shim": "*",
    "typescript": "^2.0.0"
  }
}
```

> **`tsconfig.json`**
```json
{}
```

> **`test.ts`**
```typescript
let x = new Set<string>();
```

After installing all dependencies with `npm install`, I call `tsc` once without arguments and once with `-p .`:
```
$ typings-test> tsc
$ typings-test> tsc -p .
test.ts(1,13): error TS2304: Cannot find name 'Set'.
```

As you can see, the latter fails because it cannot find a definition for `Set`. So the types that are located in `node_modules/@types/es6-shim/index.d.ts` are not loaded in this case.

I'm not the original reporter of this issue, but here's a minimal example as a gist that can be easily cloned:

https://gist.github.com/Gekkio/720511fa701e7903ef48adc6e4898e39

I ran into this issue myself with 2.0.0 beta, but this example project reproduces the issue with the latest nightly build. Here's an example run on my machine:

```
$ git clone https://gist.github.com/720511fa701e7903ef48adc6e4898e39.git
Cloning into '720511fa701e7903ef48adc6e4898e39'...
remote: Counting objects: 5, done.
remote: Compressing objects: 100% (5/5), done.
remote: Total 5 (delta 0), reused 0 (delta 0), pack-reused 0
Unpacking objects: 100% (5/5), done.
Checking connectivity... done.
$ cd 720511fa701e7903ef48adc6e4898e39
$ npm i
ts-9785@0.0.0 /home/joonas/720511fa701e7903ef48adc6e4898e39
├── @types/node@4.0.29
└── typescript@2.1.0-dev.20160718
$ tsc
$ tsc -p .
main.ts(1,23): error TS2307: Cannot find module 'path'.
```

--- a/src/compiler/tsc.ts
+++ b/src/compiler/tsc.ts
@@ -385,7 +385,8 @@ namespace ts {
                 sys.exit(ExitStatus.DiagnosticsPresent_OutputsSkipped);
                 return;
             }
-            const configParseResult = parseJsonConfigFileContent(configObject, sys, getNormalizedAbsolutePath(getDirectoryPath(configFileName), sys.getCurrentDirectory()), commandLine.options, configFileName);
+            const cwd = sys.getCurrentDirectory();
+            const configParseResult = parseJsonConfigFileContent(configObject, sys, getNormalizedAbsolutePath(getDirectoryPath(configFileName), cwd), commandLine.options, getNormalizedAbsolutePath(configFileName, cwd));
             if (configParseResult.errors.length > 0) {
                 reportDiagnostics(configParseResult.errors, /* compilerHost */ undefined);
                 sys.exit(ExitStatus.DiagnosticsPresent_OutputsSkipped);

--- a/tests/baselines/reference/umdGlobalMerge.errors.txt
+++ b/tests/baselines/reference/umdGlobalMerge.errors.txt
@@ -1,0 +1,20 @@
+tests/cases/compiler/b.d.ts(2,20): error TS2305: Module '"tests/cases/compiler/a".ns' has no exported member 'IFoo'.
+
+
+==== tests/cases/compiler/a.d.ts (0 errors) ====
+    export = ns;
+    
+    export as namespace ns;
+    
+    declare namespace ns {
+      export var x: number;
+      export interface IFoo { }
+    }
+    
+==== tests/cases/compiler/b.d.ts (1 errors) ====
+    declare namespace ns.something {
+      export var p: ns.IFoo;
+                       ~~~~
+!!! error TS2305: Module '"tests/cases/compiler/a".ns' has no exported member 'IFoo'.
+    }
+    

--- a/tests/cases/compiler/umdGlobalMerge.ts
+++ b/tests/cases/compiler/umdGlobalMerge.ts
@@ -1,0 +1,14 @@
+// @filename: a.d.ts
+export = ns;
+
+export as namespace ns;
+
+declare namespace ns {
+  export var x: number;
+  export interface IFoo { }
+}
+
+// @filename: b.d.ts
+declare namespace ns.something {
+  export var p: ns.IFoo;
+}

